assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/dateEntityMainMenuItem/CopyDate.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { pick } from 'ramda';
+import { __ } from '@wordpress/i18n';
+
+import { Copy } from '@application/ui/layout/entityActionsMenu/entityMenuItems';
+import { useDatetimeContext } from '@edtrHooks/index';
+import { useDatetimeItem } from '@edtrServices/apollo/queries';
+import { useDatetimeMutator } from '@edtrServices/apollo/mutations';
+import { useRelations } from '@appServices/apollo/relations';
+
+const CopyDate = (props) => {
+	const onClick = () => {
```

Comment:
@tn3rb I've got this error, had to revert:

![image](https://user-images.githubusercontent.com/1477453/77091168-df80c280-6a10-11ea-97b2-f997f2702a00.png)

Reply:
oh dang it... right... i always forget about that with hooks


... found a possible solution here: https://blog.logrocket.com/frustrations-with-react-hooks/

scroll down to the "**Return an executable function from the Hook**" heading (no anchors to link to)

Reply:
I've tried the solution proposed there:

![image](https://user-images.githubusercontent.com/1477453/77106270-7a849700-6a27-11ea-94f6-eb72dc6ad59e.png)

***
but still got that error:

![image](https://user-images.githubusercontent.com/1477453/77106332-95570b80-6a27-11ea-8169-20edd873cb67.png)

Reply:
What we can do instead is to render an actual component with that hook logic upon `onClick` event, such that component will return null, but still allows hook execution in function body.
Anyway we need a global pattern/ solution not just for this specific action, but for other buttons coupled to hooks as well. Let's pull this issue out of this PR.

Reply:
ok just add this functionality in with the hooks all being called at the top of the button component
and we'll create an issue for researching solutions and update later.